src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -142,13 +142,42 @@ private static void VerifyAnalyzersArgumentForStaticApis(ImmutableArray<Diagnost
                 throw new ArgumentException(CodeAnalysisResources.ArgumentElementCannotBeNull, nameof(analyzers));
             }
 
-            if (analyzers.HasDuplicates())
+            if (HasDuplicateAnalyzers(analyzers))
             {
                 // Has duplicate analyzer instances.
                 throw new ArgumentException(CodeAnalysisResources.DuplicateAnalyzerInstances, nameof(analyzers));
             }
         }
 
+        private static bool HasDuplicateAnalyzers(ImmutableArray<DiagnosticAnalyzer> analyzers)
```

Comment:
Why can't we just update `HasDuplicates()` to use a pooled hashset?

Reply:
The concern I had with doing that was that it could create a pool that the caller didn't intentionally mean to create.

Reply:
Most of our extensions like this do, I don't think this would be any different.

Reply:
This is slightly different in that the default pooledhashset pool assumes a default IEqualityComparer. One of the remaining callers of HasDuplicates passes in a custom comparer.

I could create a 2nd version of HasDuplicates in the extensions that doesn't take in the comparer and can use the default pooledhashset if you prefer.

Reply:
Went ahead and made the change such that HasDuplicates has two extension methods, depending on whether a custom comparer is used. If one isn't used, then the code utilizes PooledHashSet.